google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseInfo.java

```diff
@@ -141,6 +155,29 @@ public Builder setDialect(Dialect dialect) {
       return this;
     }
 
+    @Override
+    public Builder setProtoDescriptors(@Nonnull byte[] protoDescriptors) {
+      Preconditions.checkNotNull(protoDescriptors);
+      this.protoDescriptors = ByteString.copyFrom(protoDescriptors);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(@Nonnull InputStream inputStream) throws IOException {
+      Preconditions.checkNotNull(inputStream);
+      byte[] byteArray = ByteStreams.toByteArray(inputStream);
+      this.protoDescriptors = ByteString.copyFrom(byteArray);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(String filePath) throws IOException {
```

Comment:
Do we have a precedence of reading file on user's behalf? What about other Google Cloud client libraries. Are we reading files in the recommended way in case there is one?

Reply:
User's can read the file on their end and pass in the byte[] or InputStream as input to the `setProtoDescriptors` method. Alternatively, we also add another overloaded method that accepts the file path as input.